src/source.cc

```diff
@@ -119,7 +119,10 @@ std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr> Source::GetMatlBids(
       m = outrecipe.empty() ? \
           Material::CreateUntracked(*bit, target->comp()) : \
           Material::CreateUntracked(*bit, context()->GetRecipe(outrecipe));
-      port->AddBid(req, m, this);
+      
+      double pref = 1.0 /CalculateUnitPrice(throughput, target->quantity());
```

Comment:
Many components of this price are constants over time for any given facility.  I wonder if we can break it down and only calculate those parts once.  Maybe all the "process costs" are constants and the "input costs" are not?  So we make 2 functions, one that we call once at initialization to calculate the constant part and store it, and then add in the input costs which each bid?

Reply:
There could be a way to update the "constant" costs if/as necessary, but not with every bid